Add /bp/search API endpoint

justinjnlim opened the pull request:
Added new API endpoint /bp/search that gives us a copy of the Language Search page (/glottolog/language).
Created new mako template to allow 'include non-English names' option is checked by default. Tests will be added in subsequent commit.

justinjnlim committed:
Add /bp/search API endpoint

Added new API endpoint /bp/search that gives us a copy of the
Language Search page (/glottolog/language).
Created new mako template to allow 'include non-English names'
option is checked by default. Tests will be added in subsequent commit.

justinjnlim commented:
@baieric there are pytest tests in glottolog3/tests/test_functional.py that call the /glottolog/languages search endpoint both on its own and with a bunch of different url parameters. do you think we should mirror every test for /bp/search that currently exists for /glottolog/language, or just add a single basic one and add more later?

 it is worth noting that i have no idea how to run the tests, if they're run automatically somewhere, etc. 👍

baieric commented on glottolog3/__init__.py:
```diff
@@ -117,6 +117,11 @@ def main(global_config, **settings):
         '/glottolog',
         views.languages,
         renderer='language/search_html.mako')
+    config.add_route_and_view(
```
nit: let's put any endpoints we add into a clearly separate block (not mixed in with the other glottolog endpoints) under a comment `# Endpoints created by UW Blueprint`

We should be doing this in general in case we want to merge changes from the original glottolog3 repo into this one without too much confusion.

We may even want to put everything we add into separate files. I'll think more about that.

justinjnlim replied:
done

baieric commented on glottolog3/templates/language/bpsearch_html.mako:
```diff
@@ -0,0 +1,83 @@
+<%inherit file="../glottolog3.mako"/>
+<%namespace name="util" file="../util.mako"/>
+<%! active_menu_item = "search" %>
```
Can we update the menu to have a new item for this page? And then change this line accordingly.

justinjnlim replied:
done

baieric reviewed (commented):
Hmmm let's add tests once we combine the search bars, so we'd only need to copy over the /glottolog?search tests. Though it doesn't look like there's that many tests to begin with, so we should consider adding our own if necessary.

Note that we're copying the /glottolog endpoint, not the /glottolog/languages endpoint here.

baieric commented on glottolog3/__init__.py:
```diff
@@ -117,6 +117,11 @@ def main(global_config, **settings):
+    config.add_route_and_view(
+        'glottolog.bpsearch',
+        '/bp/search',
+        views.languages,
```
In a future PR we'll want to create a new view for this.

justinjnlim replied:
noted

justinjnlim committed:
Added menu item for bpsearch and isolated BP code

Added menu item for bpsearch.
Isolated Blueprint code additions to __init__.py with appropriate commenting.

justinjnlim commented:
@baieric made requested changes, awaiting LGTM 👍

baieric commented:
LGTM 🎉
